app/main.py

```diff
@@ -1 +1,38 @@
-# write your code here
+from typing import Union
+
+
+class Animal:
+    alive = []
+
+    def __init__(
+            self,
+            name: str,
+            health: int = 100,
+            hidden: bool = False) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = hidden
+        self.alive.append(self)
+
+    @classmethod
+    def remove_dead_animal(cls) -> None:
+        cls.alive = [animal for animal in cls.alive if animal.health > 0]
+
+    def __repr__(self) -> str:
+        return (f"{{Name: {self.name}, "
+                f"Health: {self.health}, "
+                f"Hidden: {self.hidden}}}")
+
+
+class Carnivore(Animal):
+
+    def bite(self, target: Union) -> None:
+        target.health -= 50 \
+            if (isinstance(target, Herbivore) and not target.hidden) else 0
+        Animal.remove_dead_animal()
```

Comment:
It would be inefficient to make a perform list comprehension and create a new list each time after the animal is deleted from the `alive`. Use `.remove(target)` to perform deletion in place and implement this logic inside of the `bite` method, delete the `remove_dead_animal` class method from the `Animal` class.